docs/DAS-Overview.md

```diff
@@ -0,0 +1,260 @@
+While performing this connection between AI agents and the knowledge bases, DAS
+provides a lot of other functionalities:
+
+* Higher level indexes stored in the DBMS
+* Query engine with pattern matching capabilities
+* Traverse engine to keep track of hypergraph traversal
+* Cache for query results
+* Scalable connection manager to connect the DAS with multiple other DASs
+
+This is why DAS is not just a Data Access Object or a database interface layer
+but rather a more complex OpenCog Hyperon's component that abstracts not only
+data modeling/access itself but also several other algorithms that are closely
+related to the way AI agents manipulates information.
+
+## DAS' Components
+
+DAS is delivered as a Python library
+[hyperon-das](https://pypi.org/project/hyperon-das/) which can be used in two
+different ways:
+
+1. To create a DAS server which is supposed to contain a knowledge base and
+provide it to many remote clients (somehow like a DBMS).
+2. To instantiate a DAS in a Python program which can store a smaller local
+knowledge base and can, optionally, connect to one or more remote DAS servers,
+exposing their contents to the local program. In this case, the local knowledge
+base can store its contents in RAM or can use a DB backend to persist it.
+
+<p align="center">
+<img src="assets/components.png" width="900"/>
+</p>
+
+Components in the DAs architecture are designed to provide the same 
+[data manipulation API](https://singnet.github.io/das-query-engine/api/das/)
+regardeless of whether it's being used locally or remotelly or, in the case of
+a local DAS, whether DB persistence is being used or not.
+
+Part of this API is delagated to __Traverse Engine__, which interacts with the
+Query Engine and the Cache to provide means to the user to traverse the
+Atomspace hypergraph. Operations like finding the links pointing from/to a
+given atom or finding atoms in the surounding neighboorhood are performed by
+this engine, which controls the pre-fetching of the surrounding atoms when a
+remote DAS is being used, in such a way that following links can be done
+quickly.
+
+The __Query Engine__ is where global queries are processed. These are queries
+for specific atoms or sets of atoms that satisfies some criteria, including
+pattern matching. When making a query, the user can specify whether only local
+atoms should be considered or whether atoms in remote DASs should be searched
+as well. If that's the case, the Query Engine connects to the remote OpenFaaS
+servers to make the queries in the remote DASs and return a answer which is a
+proper combination of local and remote information. For instance, if there're
+different versions of the same atom in local and one of the remote DASs, the
+local version is returned.
+
+Both engines use the __Cache__ in order to make queries involving a remote DAS
+faster. The DAS' cache is not exactly like a traditional memory or disc cache,
```

Comment:
Remove apostrophe after `DAS`

Reply:
This is grammar. `DAS' cache` means `Cache of DAS`